src/core/tests/type_prop/eye.cpp

```diff
@@ -0,0 +1,276 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "gtest/gtest.h"
+#include "ngraph/ngraph.hpp"
+#include "ngraph/opsets/opset9.hpp"
+#include "type_prop.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+TEST(type_prop, eye_constant) {
```

Comment:
Add a test where batch shape is being taken via `Parameter(PartialShape({1, 10}, {10, 25}) -> ShapeOf -(last input)-> Eye`
Output should contain `{1, 10}, {10, 25}` as batch.

Reply:
Cool test case! Added, thanks!